### PR TITLE
Don't retry celery tasks in local

### PR DESCRIPTION
**Pull request by AurelienGasser**
Companion PR: https://github.com/SubstraFoundation/substra-tests/pull/149

Retrying celery tasks once has cons:

- it usually brings more confusion than anything, because errors are reported twice 
- errors are rarely (if ever) fixed the 2nd time ﻿they are tried
- it makes the tests fail slower

This PR removes the celery retry on the local environment

### CODE REVIEW

**Comment by Kelvin-M** [comment]
Good idea, but some e2e tests check that the retry mechanism works !

**Comment by AurelienGasser** [comment]
Looks like this could be a good opportunity to reimplement this test in substra-backend

https://github.com/SubstraFoundation/substra-tests/blob/d0769d9add35323f8d4ad1864918aee96473a478/tests/test_system.py#L22-L32

**Comment by AurelienGasser** [comment]
@Kelvin-M I added a unit test, mind reviewing again ?

**Review by Kelvin-M — APPROVED** [review]

File: backend/substrapp/tests/tests_tasks.py [hunk 1]
```diff
@@ -58,6 +58,20 @@ def setUpTestData(cls):
...
         shutil.rmtree(MEDIA_ROOT, ignore_errors=True)
 
+    def test_celery_retry(self):
+        subtuple_key = 'test_owkin'
+        subtuple = {'key': subtuple_key, 'in_models': None}
+
+        with mock.patch('substrapp.tasks.tasks.do_task') as mdo_task,\
+                mock.patch('substrapp.tasks.tasks.ComputeTask.retry') as mretry:
+
+            mdo_task.side_effect = Exception('Test')
```
Comment (inalgnu) [inline backend/substrapp/tests/tests_tasks.py:68]:
```suggestion
            mdo_task.side_effect = Exception('An exeption that should trigger retry mechanism')
```

**Review by inalgnu — APPROVED** [review]

**Pull request merged**